packages/labs/analyzer/src/lib/model.ts

```diff
@@ -304,33 +403,64 @@ export type ClassHeritage = {
 export interface ClassDeclarationInit extends DeclarationInit {
   node: ts.ClassDeclaration;
   getHeritage: () => ClassHeritage;
+  fieldMap?: Map<string, ClassField> | undefined;
+  methodMap?: Map<string, ClassMethod> | undefined;
 }
 
 export class ClassDeclaration extends Declaration {
   readonly node: ts.ClassDeclaration;
   private _getHeritage: () => ClassHeritage;
   private _heritage: ClassHeritage | undefined = undefined;
+  readonly _fieldMap: Map<string, ClassField> | undefined;
+  readonly _methodMap: Map<string, ClassMethod> | undefined;
 
   constructor(init: ClassDeclarationInit) {
     super(init);
     this.node = init.node;
     this._getHeritage = init.getHeritage;
+    this._fieldMap = init.fieldMap;
+    this._methodMap = init.methodMap;
   }
 
   get heritage(): ClassHeritage {
     return (this._heritage ??= this._getHeritage());
   }
+
+  get fields() {
+    return Array.from(this._fieldMap?.values() ?? []);
```

Comment:
Do you not want to just return the iterator or undefined and let the caller convert to an array if needed? 
```suggestion
    return this._fieldMap?.values();
```

Reply:
Uh, sure. Done.